web-admin/src/features/projects/environment-variables/utils.ts

```diff
@@ -0,0 +1,77 @@
+import { EnvironmentType, type VariableNames } from "./types";
+
+/**
+ *
+ * - Environment it belongs to ("prod" or "dev").
+ * - If empty, then the value is used as the fallback for all environments.
+ */
+export function getEnvironmentLabel(environment: string) {
+  return environment === EnvironmentType.UNDEFINED ? "" : environment;
+}
+
+/**
+ * Checks if a key would create a duplicate environment variable.
+ *
+ * Rules for environment variables:
+ * 1. A key can only be tied to one environment configuration
```

Comment:
Nit: a key *can* be tied to multiple environment configurations, e.g. `my_key[dev]` & `my_key[prod]`